app/javascript/src/LoginPage/loginForms/Login3scaleForm.jsx

```diff
@@ -18,73 +23,64 @@ type Props = {
 type State = {
   username: string,
   password: string,
-  isValidUsername: ?boolean,
-  isValidPassword: ?boolean
+  validation: {
+    username: ?boolean,
+    password: ?boolean
+  }
 }
 
 class Login3scaleForm extends React.Component<Props, State> {
-  constructor (props: Props) {
-    super(props)
-    this.state = {
-      username: '',
-      password: '',
-      isValidUsername: undefined,
-      isValidPassword: undefined
-    }
-  }
-
-  setIsValidUsername = () => {
-    this.setState({isValidUsername: this.state.username !== ''})
+  state = {
+    username: '',
+    password: '',
+    validation: {}
   }
 
-  handleTextInputUsername = (username: string) => {
-    this.setState({ username }, this.setIsValidUsername)
-  }
-
-  setIsValidPassword = () => {
-    this.setState({isValidPassword: this.state.password !== ''})
+  handleInputChange = (value: string, event: SyntheticEvent<HTMLInputElement>) => {
```

Comment:
Here you could follow the same approach as I mentioned in the validateAll when submitting the form